src/manifest/extensions/ExtensionGuards.ts

```diff
@@ -0,0 +1,229 @@
+import { ShellExtensionType } from './shell/ShellExtensionType';
+import { ApplicationShellExtension } from './shell/types/ApplicationShellExtension';
+import { CompanionShellExtension } from './shell/types/CompanionShellExtension';
+import { KnowledgeShellExtension } from './shell/types/KnowledgeShellExtension';
+import { SidekickShellExtension } from './shell/types/SidekickShellExtension';
+import { ToolShellExtension } from './shell/types/ToolShellExtension';
+import { TabExtensionType } from './tabs/TabExtensionType';
+import { AccountTabExtension } from './tabs/types/AccountTabExtension';
+import { OpportunityTabExtension } from './tabs/types/OpportunityTabExtension';
+import { ProspectActionTabExtension } from './tabs/types/ProspectActionTabExtension';
+import { ProspectTabExtension } from './tabs/types/ProspectTabExtension';
+import { ReportsTabExtension } from './tabs/types/ReportsTabExtension';
+import { TileExtension } from './tiles/TileExtension';
+import { TileExtensionType } from './tiles/TileExtensionType';
+import { AccountTileExtension } from './tiles/types/AccountTileExtension';
+import { HomeEmailsTileExtension } from './tiles/types/HomeEmailsTileExtension';
+import { HomeTasksTileExtension } from './tiles/types/HomeTasksTileExtension';
+import { OpportunityTileExtension } from './tiles/types/OpportunityTileExtension';
+import { ProspectTileExtension } from './tiles/types/ProspectTileExtension';
+
+export const isProspectTileExtension = (extension: any): extension is ProspectTileExtension => {
+  if (!Object.prototype.hasOwnProperty.call(extension, 'type') || typeof extension.type !== 'string') {
```

Comment:
All of the new functions are identical up to the string value type is checked against.
Would be nice to have a private function reused for them